packages/codecs-core/src/combine-codec.ts

```diff
@@ -31,14 +32,19 @@ export function combineCodec<From, To extends From = From>(
     encoder: Encoder<From>,
     decoder: Decoder<To>
 ): Codec<From, To> {
-    if (encoder.fixedSize !== decoder.fixedSize) {
+    if (isFixedSizeCodec(encoder) !== isFixedSizeCodec(decoder)) {
+        // TODO: Coded error.
+        throw new Error(`Encoder and decoder must either both be fixed-size or variable-size.`);
+    }
```

Comment:
Sorry yep, meant to link back to the earlier comment!

Reply:
I'm gonna put a pin on that one because, since we've removed `{ fixedSize: null }` in this PR on variable-size codecs, we know have:

```ts
{} as FixedSizeDecoder<T> satisfies VariableSizeDecoder<T>
```

Which means, we get this weird behaviour in the type tests:

![CleanShot 2023-11-30 at 16 34 01@2x](https://github.com/solana-labs/solana-web3.js/assets/3642397/64def472-eac3-474c-a331-7cbae1665eb3)

Also, making this change makes us having to refactor most internal usage of `combineCodec` and do stuff like:

```ts
return isFixedSize(encoder) && isFixedSize(decoder)
  ? combineCodec(encoder, decoder)
  : combineCodec(encoder, decoder)
```

Just so we can please TypeScript.

Reply:
Ah I didn't think about that sub-type-ish relationship, I was thinking about them as discrete types. Makes sense to leave it as-is!